Complete service migration

TEdwardJ opened the pull request.

Greengrer committed:
Refactored DefaultPasswordRecoveryService and DefaultPasswordRecoveryServiceTest.

Greengrer committed:
Changed JdbcPasswordRecoveryDao and tests, fixed DefaultPasswordRecoveryServiceTest with Ted's help.
1. Changed JdbcPasswordRecoveryDao and test to account for all possible cases.
2. Changed DefaultPasswordRecoveryTest to replace beans with mock beans in application context and refactored tests.
3. Changed DefaultPasswordRecoveryService to catch specific data access exceptions and react to them.

Greengrer committed:
Changed controllers, user dao tests and user service, unfinished user service tests.
1. Changed SignUp-, SignIn-, Confirmation-, and Profile- controllers.
2. Changed UserService and implementation.
3. Added methods to security service and implementation to validate passwords and email.
4. Uncommented and started changing UserServiceTest, but haven't finished yet.
5. Added tests to SecurityServiceTest.

TEdwardJ committed:
Service migration:
 - OneVinylOfferController was simplified, since all the logic was moved into the new service OneVinylOffersServiceImpl;
 - Jdbc repositories and their interfaces for UniqueVinyl and for Offer were restored from previous commits;
 - RowMappers for UniqueVinyl and for Offer were restored from previous commits;

TEdwardJ committed:
Service migration:
 - OneVinylOffersServiceImpl decomposition of logic is made;

TEdwardJ committed:
Merge branch 'complete_service_migration' of https://github.com/littlepony99/wax-deals into complete_service_migration

TEdwardJ committed:
Service migration:
 - OneVinylOffersServiceImpl.getActualOffer() modifier changes;

TEdwardJ committed:
Service migration:
 - OneVinylOffersServiceImplTest class added  for testing of OneVinylOffersServiceImpl service;
 - OneVinylOffersServiceImpl reworked in order to make the code more concise;
 - the OfferService interface has got new method getActualizedOffer as well as its implementation;

Greengrer committed:
Refactored user service, security service, and confirmation service.
1. Refactored UserService, implementation, and tests.
2. Added password and email validation methods to security service.
3. Removed user service from confirmation service.

Greengrer committed:
Added regex for email validation

Greengrer committed:
Merge branch 'complete_service_migration' of https://github.com/littlepony99/wax-deals into complete_service_migration

TEdwardJ committed:
Merge branch 'complete_service_migration' of https://github.com/littlepony99/wax-deals into complete_service_migration

Greengrer committed:
Refactored EmailConfirmationService, Dao, and tests, changed controllers, deleted specific exceptions.
1. Refactored ConfirmationTokenDao to throw Runtime exception with specific messages.
2. Refactored DefaultEmailConfirmationService(previously DefaultConfirmationService).
3. Refactored DefaultEmailConfirmationServiceTest to be SringBootTest.
4. Deleted UserServiceException and PasswordRecoveryException.
5. Moved logic away from Profile controller and EmailConfirmation controller to services.
6. Moved method confirmSignIn from ConfirmationController to ProfileController.

littlepony99 committed:
Complete service migration. UserPostService+Captcha

TEdwardJ committed:
Complete service migration, UniqueVinyl, Offer, ElasticSearch:
 - UniqueVinyl and Offer entities are stored in ElasticSearch DB from now on;
 - the appropriate repositories migrated from add_elasticsearch branch;
 - ElasticSearch test configuration class migrated from add_elasticsearch branch;
 - TestContainers dependencies added to the project, it is used to provide elasticSearch and Postgres DBs during tests execution;
 - CurrencyConverters are migrated from add_elasticsearch branch;
 - Jdbc related repositories for UniqueVinyl and Offer entities with related classes were eliminated;

Greengrer committed:
Removed accidental video file from the project

Greengrer committed:
Merge branch 'complete_service_migration' of https://github.com/littlepony99/wax-deals into complete_service_migration

TEdwardJ committed:
Complete service migration:
 - test application.properties has got settings for postgres and elasticsearch DBs connections;

TEdwardJ committed:
Merge branch 'complete_service_migration' of https://github.com/littlepony99/wax-deals into complete_service_migration

TEdwardJ committed:
Complete service migration:
 - ConfirmationTokenResultSetExtractorTest: package path discrepancy fixed;

littlepony99 committed:
fix review comments

littlepony99 committed:
Merge branch 'serviceMigration_userPostService+Captcha' into complete_service_migration

TEdwardJ committed:
Complete service migration:
 - DefaultUserPostService.addUserPostWithCaptchaRequest() - if-condition reverted in order to meet tests expectations;

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/exception/entity/PasswordRecoveryError.java:
```diff
@@ -5,7 +5,7 @@
 @AllArgsConstructor
 @Getter
-public enum ErrorRecoveryPassword {
+public enum PasswordRecoveryError {
```
does it make sense to name it PasswordRecoveryError**s** ?

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/exception/entity/PasswordRecoveryError.java:
```diff
@@ -15,6 +15,7 @@
 
     TOKEN_NOT_CORRECT_UUID("Your link is incorrect! Please check the link in the your email"),
     TOKEN_NOT_FOUND_IN_DB("Your link is incorrect! Please check the link in the your email or contact support."),
+    DELETE_TOKEN_ERROR("Sorry, problem with processing this link. Please, request new one or contact support."),
```
I strongly believe we should have consistency here, either all constants with _ERROR suffix or none of them

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/PasswordRecoveryService.java:
```diff
@@ -1,10 +1,10 @@
 public interface PasswordRecoveryService {
 
-    void changePassword(UserChangeProfileInfoRequest userProfileInfo);
+    void changePassword(UserInfoRequest userProfileInfo);
```
Are you sure it must be "userProfileInfo" not "userInfoRequest" or just "request" ?

hemiSRT8 replied:
Could you please fix this name also in the interface ?

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/DefaultPasswordRecoveryService.java:
```diff
@@ -23,88 +23,106 @@
         }
     }
 
     void sendEmailWithLink(String email, String recoveryToken) {
```
Should have "private" access modfiier. Never consider things like "it could be helpful when...". For the moment it is being used only inside this class

TEdwardJ replied:
it is used in tests

hemiSRT8 replied:
Why dont we test general method? We can test utility methods threw it 

For instance:

changePassword is using stringToUUID

You could create test case where during execution fo changePassword other method like stringToUUID should throw exception

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/DefaultPasswordRecoveryService.java:
```diff
@@ -23,88 +23,106 @@
+        }
     }
 
     void checkPassword(String newPassword, String confirmPassword) {
```
Should have "private" access modfiier. Never consider things like "it could be helpful when...". For the moment it is being used only inside this class

TEdwardJ replied:
it is used in tests too

hemiSRT8 replied:
Why dont we test general method? We can test utility methods threw it

For instance:

changePassword is using stringToUUID

You could create test case where during execution fo changePassword other method like stringToUUID should throw exception

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/DefaultPasswordRecoveryService.java:
```diff
@@ -23,88 +23,106 @@
     }
 
-    UUID stringToUUD(String token) {
+    UUID stringToUUID(String token) {
```
Should have "private" access modfiier. Never consider things like "it could be helpful when...". For the moment it is being used only inside this class

TEdwardJ replied:
it is used in tests too

hemiSRT8 replied:
Why dont we test general method? We can test utility methods threw it

For instance:

changePassword is using stringToUUID

You could create test case where during execution fo changePassword other method like stringToUUID should throw exception

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/DefaultPasswordRecoveryService.java:
```diff
@@ -23,88 +23,106 @@
     }
 
-    void checkForNotNull(String value, ErrorRecoveryPassword emptyValue) {
+    void checkForIsNotEmptyNotNull(String value, PasswordRecoveryError emptyValue) {
```
Should have "private" access modfiier. Never consider things like "it could be helpful when...". For the moment it is being used only inside this class

TEdwardJ replied:
it is used in tests too

hemiSRT8 replied:
Why dont we test general method? We can test utility methods threw it

For instance:

changePassword is using stringToUUID

You could create test case where during execution fo changePassword other method like stringToUUID should throw exception

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/DefaultPasswordRecoveryService.java:
```diff
@@ -23,88 +23,106 @@
     }
 
-    RecoveryToken addRecoveryUserToken(long userId) {
+    RecoveryToken addRecoveryTokenWithUserId(long userId) {
```
Should have "private" access modfiier. Never consider things like "it could be helpful when...". For the moment it is being used only inside this class

TEdwardJ replied:
it is used in tests

hemiSRT8 replied:
Why dont we test general method? We can test utility methods threw it

For instance:

changePassword is using stringToUUID

You could create test case where during execution fo changePassword other method like stringToUUID should throw exception

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/DefaultPasswordRecoveryService.java:
```diff
@@ -23,88 +23,106 @@
         recoveryToken.setUserId(userId);
         recoveryToken.setToken(token);
-        passwordRecoveryDao.add(recoveryToken);
+        try {
```
You could catch DataIntegrityViolationException only here

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/DefaultPasswordRecoveryService.java:
```diff
@@ -23,88 +23,106 @@
+    private final UserService userService;
+    private final MailSender mailSender;
 
+
```
No need of extra line breaks here

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/DefaultPasswordRecoveryService.java:
```diff
@@ -23,88 +23,106 @@
+    @Transactional
     @Override
-    public void changePassword(UserChangeProfileInfoRequest userProfileInfo) {
+    public void changePassword(UserInfoRequest userProfileInfo) {
```
'userInfoRequest" or "request"

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/DefaultPasswordRecoveryService.java:
```diff
@@ -23,88 +23,106 @@
-    private final PasswordRecoveryDao passwordRecoveryDao;
-    private final UserService userService;
-    private final MailSender mailSender;
     @Value("${application.link}")
```
You could use static constants for such declarations
Actually the point of @Value is to have its actual value specified in a property file
You could also use : operator inside @Value so that default value will be used if specified key does not exist in a property file

`@Value("${application.link:default-url}")`

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/DefaultPasswordRecoveryService.java:
```diff
@@ -23,88 +23,106 @@
-    private final MailSender mailSender;
     @Value("${application.link}")
     private String applicationLink;
     @Value("${recoveryToken.live.hours}")
```
You could use static constants for such declarations
Actually the point of @value is to have its actual value specified in a property file
You could also use : operator inside @value so that default value will be used if specified key does not exist in a property file

@Value("${recoveryToken.live.hours:default-value}")

TEdwardJ replied:
The default value (24) is specified for the case when there is no appropriate application property.

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/DefaultPasswordRecoveryService.java:
```diff
@@ -23,88 +23,106 @@
-@PropertySource("classpath:application.properties")
 public class DefaultPasswordRecoveryService implements PasswordRecoveryService {
     private static final String RECOVERY_MESSAGE = "Hello, to change your password, follow this link: \n{applicationLink}/recoveryPassword/newPassword?token=";
-
```
Could you please add line breaks here? So that blocks of fields are logically separated

`private static final String CONSTANT_FIELD = "value";

@Value("${some.value}" 
private String propertyValue;

private final Service service;`

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/DefaultOfferService.java:
```diff
@@ -1,38 +1,102 @@
 
+    private final OfferRepository offerRepository;
+    private final UniqueVinylRepository uniqueVinylRepository;
+
```
no need of extra line break

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/DefaultPasswordRecoveryService.java:
```diff
@@ -23,88 +23,106 @@
 
-    void checkForNotNull(String value, ErrorRecoveryPassword emptyValue) {
+    void checkForIsNotEmptyNotNull(String value, PasswordRecoveryError emptyValue) {
         if (value == null || value.isEmpty()) {
```
apache StringUtils.isBlank(value)

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/DefaultUniqueVinylService.java:
```diff
@@ -1,5 +1,6 @@
 package com.vinylteam.vinyl.service.impl;
```
Can we remove "impl" package ?

TEdwardJ replied:
Do you mean to put all service implementations and interfaces into one package com.vinylteam.vinyl.service?

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/DefaultOfferService.java:
```diff
@@ -1,38 +1,102 @@
+            throw new IllegalArgumentException("uniqueVinylId is null");
+        }
+        List<Offer> offers = offerRepository.findByUniqueVinylId(uniqueVinylId);
+        log.debug("Resulting list of vinyls is {'vinyls':{}}", offers);
```
Do you really need to log entire list? Maybe just size?

TEdwardJ replied:
could you please advise whether debug level is not enough to show debug information in details?

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/ElasticSearchConfiguration.java:
```diff
@@ -0,0 +1,67 @@
+public class ElasticSearchConfiguration extends AbstractElasticsearchConfiguration {
+
+    @Value("${spring.data.elasticsearch.cluster-nodes:localhost}")
+    private String elasticsearchNode;
```
not used

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/ElasticSearchConfiguration.java:
```diff
@@ -0,0 +1,67 @@
+    private String elasticsearchUserPassword;
+
+    @Value("${ELASTICSEARCH_HOST:}")
+    private String awsElasticsearch;
```
not used

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/dao/OfferDao.java:
```diff
@@ -0,0 +1,14 @@
+
+public interface OfferDao {
+
+    List<Offer> findManyByUniqueVinylId(long uniqueVinylId);
```
without "Many"

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/dao/OfferDao.java:
```diff
@@ -0,0 +1,14 @@
+
+import java.util.List;
+
+public interface OfferDao {
```
Not used class

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/dao/UniqueVinylDao.java:
```diff
@@ -0,0 +1,22 @@
+
+    UniqueVinyl findById(long id);
+
+    List<UniqueVinyl> findManyRandom(int amount);
```
without "Many"

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/dao/UniqueVinylDao.java:
```diff
@@ -0,0 +1,22 @@
+
+    List<UniqueVinyl> findManyByArtist(String artist);
+
+    UniqueVinyl updateSetHasOffersFalse(UniqueVinyl vinyl);
```
Just "updateUniqueVinyl". It is up to you what you will do under the hood

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/dao/elasticsearch/UniqueVinylRepositoryCustomImpl.java:
```diff
@@ -0,0 +1,76 @@
+
+    @Override
+    public List<UniqueVinyl> findManyRandom(int amount) {
+
```
obsolete line break

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/exception/entity/EmailConfirmationError.java:
```diff
@@ -0,0 +1,21 @@
+
+@AllArgsConstructor
+@Getter
+public enum EmailConfirmationError {
```
EmailConfirmationError**s**

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/exception/entity/UserError.java:
```diff
@@ -0,0 +1,29 @@
+
+@AllArgsConstructor
+@Getter
+public enum UserError {
```
UserError**s**

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/security/SecurityService.java:
```diff
@@ -8,4 +8,10 @@
 
+    void validatePassword(String password, String confirmationPassword);
+
+    void passwordFormatCheck(String password);
```
validatePassword (polymorphism :) )

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/security/SecurityService.java:
```diff
@@ -8,4 +8,10 @@
 
     boolean checkPasswordAgainstUserPassword(User user, char[] password);
```
Lets have consistency here. Either "validate" prefix or "check"
Maybe it would be better to create separate validator class

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/security/impl/DefaultSecurityService.java:
```diff
@@ -31,7 +33,7 @@ public DefaultSecurityService() {
         } catch (NoSuchAlgorithmException e) {
             log.error("Error during initialisation of secretKeyFactory", e);
-            throw new RuntimeException(e);
+            throw new java.lang.RuntimeException(e);
```
Do you really need explicit package declaration of java.lang. here ?

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/UserService.java:
```diff
@@ -1,29 +1,27 @@
-    Optional<User> editProfile(UserChangeProfileInfoRequest userProfileInfo,
-                               User user,
-                               ModelAndView modelAndView);
+    Optional<User> editProfile(UserInfoRequest userProfileInfo,
```
Obsolete line break

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/converter/CurrencyToStringConverter.java:
```diff
@@ -0,0 +1,20 @@
+
+    @Override
+    public String convert(Optional<Currency> source) {
+        if (source.isEmpty()) {
```
return source.isEmpty() ? "" : source.get().getSymbol();

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/dao/elasticsearch/UniqueVinylRepositoryCustomImpl.java:
```diff
@@ -0,0 +1,76 @@
+@Repository
+@RequiredArgsConstructor
+@Slf4j
+public class UniqueVinylRepositoryCustomImpl implements UniqueVinylRepositoryCustom {
```
Maybe DefaultUniqueVinylRepository ?

TEdwardJ replied:
"Custom" should be in the name, so what about CustomUniqueVinylRepository?

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/dao/elasticsearch/UniqueVinylRepositoryCustomImpl.java:
```diff
@@ -0,0 +1,76 @@
+@RequiredArgsConstructor
+@Slf4j
+public class UniqueVinylRepositoryCustomImpl implements UniqueVinylRepositoryCustom {
+    private static final String INDEX = "unique_vinyl_index";
```
Please add line break

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/dao/elasticsearch/UniqueVinylRepositoryCustomImpl.java:
```diff
@@ -0,0 +1,76 @@
+                .withQuery(queryBuilder)
+                .build();
+
+        List<UniqueVinyl> vinylMatches = getUniqueVinyls(searchQuery);
```
Obsolete variable declaration, just return

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/dao/elasticsearch/UniqueVinylRepositoryCustomImpl.java:
```diff
@@ -0,0 +1,76 @@
+    @Override
+    public List<UniqueVinyl> findManyRandom(int amount) {
+
+        FunctionScoreQueryBuilder functionScore = QueryBuilders.functionScoreQuery(
```
Obsolete line break

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/entity/UniqueVinyl.java:
```diff
@@ -1,91 +1,41 @@
-    }
+    @Field(type = FieldType.Boolean)
+    @EqualsAndHashCode.Exclude
+    private boolean hasOffers;
```
Obsolete line breaks below hasOffers

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/entity/UniqueVinyl.java:
```diff
@@ -1,91 +1,41 @@
+
+@Data
+@Document(indexName = "unique_vinyl_index")
 public class UniqueVinyl {
```
Why do we need both getHasOffers and isHasOffers methods?

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/security/impl/DefaultSecurityService.java:
```diff
@@ -74,7 +98,7 @@ String hashPassword(char[] password, byte[] salt, int iterations) {
         } catch (InvalidKeySpecException e) {
             log.error("Error during hashing password", e);
-            throw new RuntimeException(e);
+            throw new java.lang.RuntimeException(e);
```
Do you really need explicit package declaration of java.lang. here ?

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/DefaultEmailConfirmationService.java:
```diff
@@ -17,10 +17,9 @@
 
     private final ConfirmationTokenDao confirmationTokenDao;
-    private final UserDao userDao;
     private final MailSender mailSender;
```
Please add extra line break here

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/OfferService.java:
```diff
@@ -9,12 +9,14 @@
 public interface OfferService {
 
-    List<Offer> findManyByUniqueVinylId(long uniqueVinylId);
+    List<Offer> findManyByUniqueVinylId(String uniqueVinylId);
```
without "Many"

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/OfferService.java:
```diff
@@ -9,12 +9,14 @@
 
     void updateUniqueVinylsRewriteAll(List<UniqueVinyl> uniqueVinyls, List<Offer> offers);
 
     List<Integer> getListOfShopIds(List<Offer> offers);
```
findShopIds

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/OfferService.java:
```diff
@@ -9,12 +9,14 @@
-    void mergeOfferChanges(Offer offer, VinylParser shopParser, RawOffer dynamicOffer);
+    Offer mergeOfferChanges(Offer offer, VinylParser shopParser, RawOffer dynamicOffer);
+
+    Offer getActualizedOffer(Offer dbOffer);
```
findActualizedOffer

TEdwardJ replied:
it does  not find actually, it actualizes the input offer, so what about actualizeOffer()?

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/OneVinylOffersServiceImpl.java:
```diff
@@ -0,0 +1,95 @@
+
+    @Override
+    public OneVinylPageFullResponse prepareOneVinylInfo(String identifier) {
+        String uniqueVinylId = identifier;
```
Obsolete variable declaration

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/OneVinylOffersServiceImpl.java:
```diff
@@ -0,0 +1,95 @@
+    private final DiscogsService discogsService;
+
+    @Override
+    public OneVinylPageFullResponse prepareOneVinylInfo(String identifier) {
```
identifier > id

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/OneVinylOffersServiceImpl.java:
```diff
@@ -0,0 +1,95 @@
+
+        uniqueVinylService.findManyByArtist(uniqueVinyl.getArtist())
+                .stream()
+                .filter(v -> v.getId() != uniqueVinyl.getId())
```
maybe compare by equals?

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/OneVinylOffersServiceImpl.java:
```diff
@@ -0,0 +1,95 @@
+                .stream()
+                .filter(store -> Objects.equals(store.getId(), offer.getShopId()))
+                .findFirst()
+                .get();
```
You dont need Optional here if you use .get() without isPresent or orElse(orElseThrow)

TEdwardJ replied:
No need in  isPresent or orElse(orElseThrow), it is just "lookup" logic, and the logic afore in stack guarantees that this method will surely find the result

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/UniqueVinylService.java:
```diff
@@ -10,7 +10,7 @@
 
```
Please remove Many word from methods names

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/UniqueVinylService.java:
```diff
@@ -10,7 +10,7 @@
 
     List<UniqueVinyl> findAll();
 
```
return value of updateOneUniqueVinylAsHavingNoOffer is never used
please consider making it void

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/DefaultUserPostService.java:
```diff
@@ -28,11 +36,32 @@ public void processAdd(UserPost post) {
     }
 
-    protected String createContactUsMessage(String recipient, String subject, String mailBody) {
+    String createContactUsMessage(String recipient, String subject, String mailBody) {
```
No need of using StringBuilder unless you concat strings in a cycle
This usage does not result in significant performance improvement

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/DefaultCaptchaService.java:
```diff
@@ -15,6 +15,7 @@
+@RequiredArgsConstructor
 public class DefaultCaptchaService implements CaptchaService {
 
     private final RestTemplate template;
```
Please add extra line break

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/DefaultCaptchaService.java:
```diff
@@ -23,10 +24,6 @@
     @Value("${google.recaptcha.secret}")
```
Please add extra line break

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/DefaultOfferService.java:
```diff
@@ -1,38 +1,102 @@
 
     @Override
     public void updateUniqueVinylsRewriteAll(List<UniqueVinyl> uniqueVinyls, List<Offer> offers) {
+        if (uniqueVinyls == null) {
```
Please replace null and isempty checks with apache CollectionUtils.isEmpty()

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/DefaultUserPostService.java:
```diff
@@ -2,22 +2,30 @@
     private final UserPostDao userPostDao;
     private final MailSender mailSender;
+    private final CaptchaService captchaService;
+    @Value("${project.mail}")
```
+ extra line break

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/DefaultUserPostService.java:
```diff
@@ -2,22 +2,30 @@
 
-    @Value("${project.mail}")
-    private String projectMail;
     private final static String CONTACT_US_DEFAULT_THEME = "Mail from customer";
```
+ extra line break

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/DefaultUserService.java:
```diff
@@ -26,159 +24,163 @@
     }
 
     @Override
     public Optional<User> findByEmail(String email) {
```
Please return actual user instead of Optional. You have already checked whether there is a value in the Optional so it does not make sense to do .get() on clients of this methods

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/impl/DefaultUserService.java:
```diff
@@ -26,159 +24,163 @@
+        return findByEmail(newEmail);
+    }
+
+    boolean isNotEmptyNotNull(String string) {
```
Apache StringUitls.isNotBlank

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/web/controller/ContactUsController.java:
```diff
@@ -17,16 +14,15 @@
     private final UserPostService userPostService;
-    private final CaptchaService defaultCaptchaService;
-    private final DefaultCaptchaService service;
+    @Value("${project.mail}")
```
+ extra line break

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/web/controller/EmailConfirmationController.java:
```diff
@@ -0,0 +1,38 @@
+@RequestMapping("/emailConfirmation")
+public class EmailConfirmationController {
+
+    private final UserService userService;
```
not used

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/web/controller/EmailConfirmationController.java:
```diff
@@ -0,0 +1,38 @@
+    @GetMapping
+    public ModelAndView getConfirmationPage(@RequestParam(value = "token") String tokenAsString) {
+        ModelAndView modelAndView = new ModelAndView();
+        Optional<ConfirmationToken> optionalConfirmationToken = emailConfirmationService.findByToken(tokenAsString);
```
token variable not used

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/web/controller/OneVinylOfferController.java:
```diff
@@ -1,110 +1,45 @@
     @GetMapping
     public String getOneVinylOfferPage(@SessionAttribute(value = "user", required = false) User user,
-                                       @RequestParam(value = "id") String stringId,
+                                       @RequestParam(value = "id") String identifier,
```
identifier > id

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/service/OneVinylOffersService.java:
```diff
@@ -0,0 +1,10 @@
+
+public interface OneVinylOffersService {
+
+    OneVinylPageFullResponse prepareOneVinylInfo(String identifier);
```
identifier > id

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/web/controller/ProfileController.java:
```diff
@@ -43,16 +43,17 @@ public ModelAndView editProfile(@RequestParam(value = "email") String newEmail,
                                     @RequestParam(value = "discogsUserName") String newDiscogsUserName,
                                     HttpSession session,
                                     @SessionAttribute("user") User user) {
```
add line break

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/web/controller/PasswordRecoveryController.java:
```diff
@@ -74,17 +73,17 @@ public ModelAndView changePassword(@SessionAttribute(value = "user", required =
                                        @RequestParam(value = "recoveryToken") String token,
                                        Model model) {
```
add line break

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/web/controller/SignInController.java:
```diff
@@ -43,31 +46,34 @@ public ModelAndView signIn(@SessionAttribute(value = "user", required = false) U
         ModelAndView modelAndView = new ModelAndView();
```
add line break

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/web/controller/SignInController.java:
```diff
@@ -43,31 +46,34 @@ public ModelAndView signIn(@SessionAttribute(value = "user", required = false) U
+                                      @RequestParam(value = "email") String email,
+                                      @RequestParam(value = "password") String password,
+                                      @Value("${session.maxInactiveInterval}") Integer sessionMaxInactiveInterval,
+                                      HttpSession session) {
```
add line break

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/web/controller/SignInController.java:
```diff
@@ -43,31 +46,34 @@ public ModelAndView signIn(@SessionAttribute(value = "user", required = false) U
+        ModelAndView modelAndView = new ModelAndView();
+        log.info("Sign in user with email {} and token {}", email, tokenAsString);
+        Optional<User> optionalUser = userService.confirmEmail(userProfileInfo);
+        User user = optionalUser.get();
```
You have already checked option content in the userService, please return actual User from there so you dont need to perform .get() everywhere where you calling confirmEmail method

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/web/controller/SignInController.java:
```diff
@@ -2,21 +2,24 @@
-@RequestMapping("/signIn")
 public class SignInController {
     private final UserService userService;
     private final Integer sessionMaxInactiveInterval;
```
not used

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/web/controller/SignUpController.java:
```diff
@@ -34,37 +34,22 @@ public ModelAndView signUpUser(@SessionAttribute(value = "user", required = fals
                                    @RequestParam(value = "discogsUserName") String discogsUserName,
                                    Model model) {
         WebUtils.setUserAttributes(user, model);
```
add line break

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/web/dto/OneVinylPageFullResponse.java:
```diff
@@ -0,0 +1,19 @@
+@AllArgsConstructor
+public class OneVinylPageFullResponse {
+
+    List<OneVinylOffersServletResponse> offersResponseList;
```
No point to add "List" suffix. It is obvious from field type

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/web/dto/OneVinylPageFullResponse.java:
```diff
@@ -0,0 +1,19 @@
+
+    List<OneVinylOffersServletResponse> offersResponseList;
+
+    List<UniqueVinyl> preparedVinylsList;
```
No point to add "List" suffix. It is obvious from field type

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/web/handler/ApiExceptionHandler.java:
```diff
@@ -1,29 +1,27 @@
-
-    @ExceptionHandler(UserServiceException.class)
-    public ModelAndView handleUserServiceException(UserServiceException exception) {
+//FIXME: Fix situation with exceptions in a smart way.
```
I understand why you commented out this block, but best practice is to create JIRA ticket with this code block and describe it appropriately so that such code can be removed from the codebase

hemiSRT8 commented on src/main/java/com/vinylteam/vinyl/web/util/WebUtils.java:
```diff
@@ -32,6 +33,11 @@ public static OneVinylOffersServletResponse getOfferResponseFromOffer(Offer offe
+        setModelContext(fullResponse.getPreparedVinylsList(), fullResponse.getOffersResponseList(), model);
+    }
+
     public static void setModelContext(List<UniqueVinyl> vinylList, List<OneVinylOffersServletResponse> vinylOffersList, Model model) {
```
No point to add "List" suffix to the vinyOffersList parameter, it is obvious from the parameter type

hemiSRT8 commented on src/test/java/com/vinylteam/vinyl/ElasticSearchTestConfiguration.java:
```diff
@@ -0,0 +1,51 @@
+            ES_CONTAINER.withCreateContainerCmdModifier(cmd -> cmd
+                    .getHostConfig()
+                    .withPortBindings(PortBinding.parse("9200:9200/tcp"))
+                    .withMemory(512000000l)
```
always use upper case for long declarations
l > L

hemiSRT8 commented on src/test/java/com/vinylteam/vinyl/ElasticSearchTestConfiguration.java:
```diff
@@ -0,0 +1,51 @@
+                    .getHostConfig()
+                    .withPortBindings(PortBinding.parse("9200:9200/tcp"))
+                    .withMemory(512000000l)
+                    .withMemorySwap(512000000l));
```
always use upper case for long declarations
l > L

hemiSRT8 commented on src/test/java/com/vinylteam/vinyl/ElasticSearchTestConfiguration.java:
```diff
@@ -0,0 +1,51 @@
+public class ElasticSearchTestConfiguration {
+
+    private static final String ELASTIC_SEARCH_DOCKER = "elasticsearch:7.12.1";
+
```
remove line break

hemiSRT8 commented on src/test/java/com/vinylteam/vinyl/ElasticSearchTestConfiguration.java:
```diff
@@ -0,0 +1,51 @@
+    private static final String ELASTIC_SEARCH_DOCKER = "elasticsearch:7.12.1";
+
+    private static final String CLUSTER_NAME = "cluster.name";
+
```
remove line break

hemiSRT8 commented on src/test/java/com/vinylteam/vinyl/WaxDealsTestContainersTest.java:
```diff
@@ -0,0 +1,29 @@
+    private ElasticsearchContainer elasticsearchContainer;
+
+    @Container
+    public PostgreSQLContainer container = WaxDealsPostgresqlContainer.getInstance();
```
Any chance to avoid raw declaration and use generic ?

TEdwardJ committed:
Complete service migration PR review:
 - empty lines, unused imports changes;
 - class that contains errors description renamed to name in plural form - UserError->UserErrors;
 - Optional usage elimination where it is not worthy;
 - changes related to naming conventions;
 - Apache common-lang3 added as dependency;
 - many others remarks;

TEdwardJ committed:
Complete service migration PR review:
 - UniqueVinylRepositoryCustom renamed to CustomUniqueVinylRepository;
 - UniqueVinylRepositoryCustomImpl renamed to CustomUniqueVinylRepositoryImpl;

littlepony99 commented on pom.xml:
```diff
@@ -197,6 +208,24 @@
+        <dependency>
+            <groupId>org.testcontainers</groupId>
+            <artifactId>testcontainers</artifactId>
+            <version>1.15.3</version>
```
version to variable plz

littlepony99 commented on pom.xml:
```diff
@@ -197,6 +208,24 @@
+        <dependency>
+            <groupId>org.testcontainers</groupId>
+            <artifactId>junit-jupiter</artifactId>
+            <version>1.15.3</version>
```
version to variable plz

littlepony99 commented on pom.xml:
```diff
@@ -197,6 +208,24 @@
+        <dependency>
+            <groupId>org.testcontainers</groupId>
+            <artifactId>elasticsearch</artifactId>
+            <version>1.15.3</version>
```
version to variable plz

littlepony99 commented on src/test/java/com/vinylteam/vinyl/service/impl/OneVinylOffersServiceImplTest.java:
```diff
@@ -0,0 +1,107 @@
+    @InjectMocks
+    private OneVinylOffersServiceImpl oneVinylService;
+
+    @Autowired
```
do we really need Autowired with mockBean?

littlepony99 commented on src/test/java/com/vinylteam/vinyl/service/impl/OneVinylOffersServiceImplTest.java:
```diff
@@ -0,0 +1,107 @@
+
+    @Autowired
+    @InjectMocks
+    private OneVinylOffersServiceImpl oneVinylService;
```
do we really need Autowired with InjectMocks?

littlepony99 commented on src/test/java/com/vinylteam/vinyl/service/impl/OneVinylOffersServiceImplTest.java:
```diff
@@ -0,0 +1,107 @@
+    private OfferService offerService;
+
+    @Autowired
+    @MockBean
```
do we really need Autowired with mockBean?

TEdwardJ committed:
Complete service migration PR review:
 - unnecessary @Autowired elimination;
 - pom.xml: version for elasticSearch and testContainers in variables;

TEdwardJ committed:
Complete service migration PR review:
 - OneVinylOffersServiceImpl renamed in accordance with general service name convention;
 - empty delimiter lines added before fields with @Value annotation;
 - calls of StringUtils.isAllEmpty() changed for calls of isBlank();
 - DefaultPasswordRecoveryService`s methods with default access modifiers have got private access modifier, so now these methods are tested implicitly, trough calls of public methods that зкусуву in the call stack;
